pkg/registry/softwarecomposition/networkneighborhood/strategy.go

```diff
@@ -62,9 +64,22 @@ func (NetworkNeighborhoodStrategy) PrepareForUpdate(ctx context.Context, obj, ol
 	newAP := obj.(*softwarecomposition.NetworkNeighborhood)
 	oldAP := old.(*softwarecomposition.NetworkNeighborhood)
 
+	// if we have an network neighborhood that is marked as complete and completed, we do not allow any updates
+	if oldAP.Annotations[helpers.CompletionMetadataKey] == helpers.Complete && oldAP.Annotations[helpers.StatusMetadataKey] == helpers.Completed {
```

Comment:
Why it's called `oldAP` and not `oldNN`?
Same for the rest.

Reply:
it was like that... at least it simplifies the diffs

Reply:
actually we should use generics here